src/refiners/foundationals/fuyu/tokenizer.py

```diff
@@ -0,0 +1,210 @@
+import gzip
+import json
+import re
+from pathlib import Path
+from typing import Dict, List, TypedDict
+
+from numpy import round
+from torch import Tensor, tensor
+
+import refiners.fluxion.layers as fl
+
+
+class TokenDict(TypedDict):
+    id: int
+    content: str
+    single_word: bool
+    lstrip: bool
+    rstrip: bool
+    normalized: bool
+    special: bool
+
+
+class FuyuTokenizer(fl.Module):
+    """
+    Implement a Unigram Tokenizer based on a vocabulary file for Fuyu
+    """
+
+    def __init__(
+        self,
+        vocabulary_path: str | Path | None = None,
+    ) -> None:
+        super().__init__()
+
+        if vocabulary_path is None:
+            vocabulary_path = Path.home() / ".cache/refiners/fuyu-8b/tokenizer.json.gz"
+            if not vocabulary_path.exists():
+                raise FileNotFoundError(f"Vocabulary file not found at {vocabulary_path}")
```

Comment:
nit, simply make the vocabulary_path required (i.e. no need to set it to None by default)

Reply:
The vocabulary path is now required